### ISSUE TITLE
Error in StyleGAN and Faster Stylegan

### ISSUE THREAD

**Issue by rosas-github** [PacktPublishing/Hands-On-Image-Generation-with-TensorFlow-2.0#16]
Hi! I'm learning stylegans and I have been investigating different books and codes implementing this type of GAN. I tried to replicate your code (from your book and this repository), but both scripts have different errors:
- StyleGAN: Data cardinality is ambiguous:
  x sizes: 16, 16, 1, 16, 16, 16, 16, 16, 16, 16
  y sizes: 16
Make sure all arrays contain the same number of samples
-> This issue runs the first set of images, but in the next training it crushes with this error. Tried to checked the sizes, but haven't found anything.

- Faster StyleGAN: len is not well defined for symbolic Tensors
-> This code doesn't even begin the training. Different searches show that the solution is downgrading tensorflow to 1.5 (currently i'm working 2.5 in google colab), but it doesn't fix the problem.

I would like to know if you had this issues as well.
Thanks for your great work.

**Comment by soon-yau** [comment]
The code has been well tested by different people:
- StyleGAN - use Tensorflow 2.3.1 or below. There is new check with TF >2.4.0 that throws error about input shape.
- Faster StyleGAN - You can use any TF 2 version but never use TF1.  That code has been adapted to be used as Keras official example in https://keras.io/examples/generative/stylegan/. The core code is similar, it is just some code refactoring and polishing. So, an alternative is for you to try the Keras one, but that code uses lower resolution Celeb-A, so the result is not as good as the one in this repo. 

If you still see problem, share your Google Colab notebook with me and I'll take a look.

**Comment by rosas-github** [comment]
Hi! Yes, I found the Keras official example and checked the error out. It was just doing the next change:
loss = tf.reduce_sum(loss, axis=np.arange(1,len(tf.shape(loss)))) -> len error
loss = tf.reduce_sum(loss, axis=tf.range(1, tf.size(tf.shape(loss)))) -> solution

So, Faster StyleGAN works perfectly, and I have been working this last two days.
The original StyleGAN indeed has a problem with the version, but I haven't checked it out.
Thank for the helping hand!
